src/mento.ts

```diff
@@ -0,0 +1,260 @@
+   */
+  async getTradeablePairs(): Promise<[Asset, Asset][]> {
+    const exchanges = await this.getExchanges()
+    let pairs: [Asset, Asset][] = []
+
+    for (let exchange of exchanges) {
+      const asset0 = exchange.assets[0]
+      const asset1 = exchange.assets[1]
+      let symbols = await Promise.all([
+        getSymbolFromTokenAddress(this.provider, asset0),
+        getSymbolFromTokenAddress(this.provider, asset1),
+      ])
+      pairs.push([
+        { address: asset0, symbol: symbols[0] },
+        { address: asset1, symbol: symbols[1] },
+      ])
+    }
+
+    return pairs
+  }
+
+  /**
+   * Returns the amount of tokenIn to be sold to buy amountOut of tokenOut
+   * @param tokenIn the token to be sold
+   * @param tokenOut the token to be bought
+   * @param amountOut the amount of tokenOut to be bought
+   * @returns the amount of tokenIn to be sold
+   */
+  async getAmountIn(
+    tokenIn: string,
+    tokenOut: string,
+    amountOut: BigNumber
+  ): Promise<BigNumber> {
+    const exchange = await this.getExchangeForTokens(tokenIn, tokenOut)
+    return await this.broker.getAmountIn(
+      exchange.providerAddr,
+      exchange.id,
+      tokenIn,
+      tokenOut,
+      amountOut
+    )
+  }
+
+  /**
+   * Returns the amount of tokenOut to be bought by selling amountIn of tokenIn
+   * @param tokenIn the token to be sold
+   * @param tokenOut the token to be bought
+   * @param amountIn the amount of tokenIn to be sold
+   * @returns the amount of tokenOut to be bought
+   */
+  async getAmountOut(
+    tokenIn: string,
+    tokenOut: string,
+    amountIn: BigNumber
+  ): Promise<BigNumber> {
+    const exchange = await this.getExchangeForTokens(tokenIn, tokenOut)
+    return await this.broker.getAmountOut(
+      exchange.providerAddr,
+      exchange.id,
+      tokenIn,
+      tokenOut,
+      amountIn
+    )
+  }
+
+  /**
+   * Executes a swap with a fixed amount of tokenIn and a minimum amount of tokenOut
+   * @param tokenIn the token to be sold
+   * @param tokenOut the token to be bought
+   * @param amountIn the amount of tokenIn to be sold
+   * @param amountOutMin the minimum amount of tokenOut to be bought
+   * @returns an ethers TransactionResponse object
+   */
+  async swapIn(
+    tokenIn: string,
+    tokenOut: string,
+    amountIn: BigNumber,
+    amountOutMin: BigNumber
+  ): Promise<TransactionResponse> {
+    if (!this.signer) {
+      throw new Error('A signer is required to execute a swap')
+    }
+
+    if (!this.signer.provider) {
+      throw new Error(
+        'The signer must be connected to a provider to execute a swap'
+      )
+    }
+
+    const exchange = await this.getExchangeForTokens(tokenIn, tokenOut)
+    const tx = await this.broker.populateTransaction.swapIn(
+      exchange.providerAddr,
+      exchange.id,
+      tokenIn,
+      tokenOut,
+      amountIn,
+      amountOutMin
+    )
+    return await this.signer.sendTransaction(tx)
```

Comment:
Unrelated, `return await` has some subtle pitfalls in javascript and you generally don't want to do it. Either `return this.signer.sendTransaction` directly which will return a promise callers can await for await it then return a value on a separate line.